src/Host.cpp

```diff
@@ -477,11 +475,112 @@ Host::~Host()
     if (mpDockableMapWidget) {
         mpDockableMapWidget->deleteLater();
     }
-    mIsGoingDown = true;
-    mIsClosingDown = true;
     mErrorLogStream.flush();
     mErrorLogFile.close();
-    TDebug::removeHost(this);
+    // We are in the destructor - so it is not a good idea for anything we call
+    // to refer to our members, given that we want the name of the profile in
+    // the following pass it in as a value so we don't need to access the member
```

Comment:
I find this hard to parse, how about this instead?

```suggestion
    // Since this is a destructor, it's risky to rely on member variables within the destructor itself.
    // To avoid this, we can pass the profile name as an argument instead of accessing it
    // directly as a member variable. This ensures the destructor doesn't depend on the 
    // object's state being valid.

```